### PR TITLE
Configuration file related changes

### PR DESCRIPTION
**Pull request by kryalama**
Enhancements:
Improve discoverability of configuration file syntax error
Fail on invalid config attributes instead of silently ignoring

### CODE REVIEW

File: agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/MainEntryPoint.java [hunk 1]
```diff
@@ -118,18 +119,35 @@ static FriendlyException getFriendlyException(Throwable t) {
...
+                Path agentPath = new File(bootstrapURL.toURI()).toPath();
+                startupLogger = configureLogging(new SelfDiagnostics(), agentPath);
+                if (isFriendlyException) {
+                    startupLogger.error(message);
+                } else {
+                    startupLogger.error(message, t);
+                }
+            } catch (URISyntaxException e) {
+                e.printStackTrace();
+            } finally {
```
Comment (trask) [inline agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/MainEntryPoint.java:142]:
```suggestion
            } catch (Throwable e) {
                e.printStackTrace();
```

File: agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/MainEntryPoint.java [hunk 2]
```diff
@@ -118,18 +119,35 @@ static FriendlyException getFriendlyException(Throwable t) {
...
+                // If the startupLogger still have some issues being initialized, just print the error stack trace
+                if(startupLogger == null) {
+                    if (isFriendlyException) {
+                        System.err.println(message);
+                    } else {
+                        t.printStackTrace();
+                    }
+                }
```
Comment (trask) [inline agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/MainEntryPoint.java:150]:
```suggestion
                if (isFriendlyException) {
                    System.err.println(message);
                } else {
                    t.printStackTrace();
                }
```

File: agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java [hunk 3]
```diff
@@ -288,7 +288,7 @@ public static Configuration loadJsonConfigFile(Path configPath) throws IOExcepti
         long lastModifiedTime = attributes.lastModifiedTime().toMillis();
         try (InputStream in = Files.newInputStream(configPath)) {
             Moshi moshi = MoshiBuilderFactory.createBuilderWithAdaptor();
-            JsonAdapter<Configuration> jsonAdapter = moshi.adapter(Configuration.class);
+            JsonAdapter<Configuration> jsonAdapter = moshi.adapter(Configuration.class).failOnUnknown();
```
Comment (trask) [inline agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java:291]: nice

**Review by trask — COMMENTED** [review]
does the user see a friendly message on invalid json attribute?

**Comment by kryalama** [comment]
> does the user see a friendly message on invalid json attribute?

I was thinking to handle this in a new PR. I will create a work item for this.

File: test/smoke/appServers/global-resources/spanprocessor_applicationinsights.json [hunk 4]
```diff
@@ -78,10 +78,7 @@
             }
           ]
         }
-      ],
-      "selfDiagnostics": {
-        "level": "debug"
-      }
+      ]
```
Comment (trask) [inline test/smoke/appServers/global-resources/spanprocessor_applicationinsights.json:81]: 😄 👍

**Review by trask — APPROVED** [review]

**Pull request merged**